Lib/asyncio/runners.py

```diff
@@ -97,7 +103,21 @@ def run(self, coro, *, context=None):
 
         if context is None:
             context = self._context
-        task = self._loop.create_task(coro, context=context)
+
+        if futures.isfuture(coro):
+            # This covers the argument being an asyncio.Future or asyncio.Task
+            task = coro
+        elif coroutines.iscoroutine(coro):
+            task = self._loop.create_task(coro, context=context)
+        elif inspect.isawaitable(coro):
```

Comment:
is there a more lightweight to check if coro is an awaitable?
just importing `inspect` is pretty heavy - perhaps defer the import only to when it's needed? (i.e. if coro is neither a coroutine nor a future nor a task)

Reply:
There's already an include of "inspect" in Lib/asyncio/tasks.py even before my changes here, and that's included by Lib/asyncio/runners.py, so I don't think trying to defer the import makes sense here. If I'm able to factor out the code I need from tasks.py into a common function, I may be able to avoid the direct import in runners.py and have only the import that already existed in tasks.py, but either way the inspect module is going to be imported.

The function `inspect.isawaitable()` also doesn't look all that expensive to run:

```python
def isawaitable(object):
    """Return true if object can be passed to an ``await`` expression."""
    return (isinstance(object, types.CoroutineType) or
            isinstance(object, types.GeneratorType) and
                bool(object.gi_code.co_flags & CO_ITERABLE_COROUTINE) or
            isinstance(object, collections.abc.Awaitable))
```

Reply:
you're right, this wouldn't be a regression since `inspect` is already imported in `tasks.py`.
the specific function may be inexpensive -- it's the overall import of the `inspect` module that's quite heavy.
we could try avoiding it in `tasks.py` too, but that would be a separate issue and PR.

Reply:
Thanks - makes sense.